### PR TITLE
add the future prognose

### PR DESCRIPTION
**Pull request by jonasBer**
# Pull Request Template

## What
closes:#108
Befor
Bildschirmfoto 2024-10-12 um 16 09 55

After
Bildschirmfoto 2024-10-12 um 16 09 44

## How
Use the total and the date range to get a avg of the last period

### CODE REVIEW

**Comment by jonasBer** [comment]
@manuelalferez if you wan't  you can review this and give me feedback

File: src/app/dashboard/components/total/forecast.tsx [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+    return futureValues;
+}
+
+export default function SummaryChart() {
+    const { filteredTransactions, currency, selected } = useContext(AppContext);
+    const [data, setData] = useState<ChartData[]>([]);
+
+    useEffect(() => {
+        console.log("Selected value updated:", selected); // Log the selected value
```
Comment (manuelalferez) [inline src/app/dashboard/components/total/forecast.tsx:51]: Please, remove all the unnecessary comments. For future occasions, I am not telling you not to rely on the AI to find more optimal solutions or improve your code, but accepting everything the AI gives you without even worrying about what it generates shows a real lack of concern

File: src/app/dashboard/components/total/forecast.tsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+            ? roundToTwoDecimal(totalExpenseValue / dateRangeInDays)
+            : 0;
+
+        const avgIncome = dateRangeInDays > 0
+            ? roundToTwoDecimal(totalIncomeValue / dateRangeInDays)
+            : 0;
+
+        // Logge die berechneten Durchschnittswerte aus
+        console.log("Avg Expense per day:", avgExpense);
+        console.log("Avg Income per day:", avgIncome);
```
Comment (manuelalferez) [inline src/app/dashboard/components/total/forecast.tsx:83]:
```suggestion
```

File: src/app/dashboard/components/total/forecast.tsx [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        console.log("selcted",selected)
```
Comment (manuelalferez) [inline src/app/dashboard/components/total/forecast.tsx:84]:
```suggestion
```

File: src/app/dashboard/components/total/forecast.tsx [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+                                }}
+                            />
+                            <YAxis
+                                tickFormatter={(number) => `${currency}${number}`}
+                                tickCount={6}
+                            />
+                            <Legend />
+                            <Tooltip content={<SummaryTooltip currency={currency} />} />
+
+                            {/* Gestrichelte Linie für Ausgaben */}
```
Comment (manuelalferez) [inline src/app/dashboard/components/total/forecast.tsx:149]: Remove all unnecessary comments, please

**Comment by manuelalferez** [comment]
I like the implementation, but I can't see the value of the forecast. The future is not always a perfect upward or downward line. It would be more interesting to draw something more realistic. For example: 
<p align="center">
image
<p/>
I would improve that part: make the prediction more based on the past, not a mere accumulation of the median of the last period. 

Knowing the income and expenses is good, but also the balance in that period. How much money will I have in the account? Please, add that balace as information. Do not add a line for that in the chart, it is not mandatory (although it could be added). Users will be more interested in knowing their balance. 
<p align="center">
image
<p/>

**Comment by manuelalferez** [comment]
After adding those small changes, you should be good to merge. Everything looks great. Thanks for implementing the feature @jonasBer!

**Pull request closed without merging**